### PR TITLE
Elasticsearch: Fix adding of adhoc filters when jumping to explore

### LINKED ISSUE #55916: Elasticsearch: When jumping from dashboard to Explore, ad-hoc filters are not being added to query
More context here: https://raintank-corp.slack.com/archives/C02TXB92G6T/p1664283234413199
The same as was fixed for Loki & Prometheus in https://github.com/grafana/grafana/pull/55915

There is a bug around AdHoc filters and the Explore experience for elasticsearch. When you have AdHoc filters (type of dashboards variable) and you jump to Explore, all variables are interpolated, but AdHoc filters are ignored. For consistency, it would make sense to interpolate and include all variables. Not just some of them. Each data source is handling this, so the fix needs to be implemented directly in ES. 

How to reproduce: 
- Apply an ad-hoc filter on a dashboard
- See a panel of results get updated, scoped more tightly by the ad-hoc filter just applied
"Explore" that panel
- Within the explore view, I want to see the same ad-hoc filters still present, and see the same results rendered

### PR DESCRIPTION
**Pull request by ivanahuckova**
The same as was fixed for Loki & Prometheus in https://github.com/grafana/grafana/pull/55915

This PR fixes a bug around adhoc filters and the Explore experience for elasticsearch. When you have adhoc filters (type of dashboards variable) and you jump to Explore, all variables are interpolated, but adhoc filters were ignored. For consistency, it would make sense to interpolate and include all variables. Not just some of them. 

In this PR, we are creating a new data source method `addAdHocVariables` and then using it in `interpolateVariablesInQuery`, which is a function used to interpolate variables when jumping from dashboard to explore.

I have also added test coverage for all the new functionality.

Lastly, I have replaced usage of `NOT` operator to `-`, as that is recommended as preferred operator: https://www.elastic.co/guide/en/elasticsearch/reference/8.5/query-dsl-query-string-query.html#_boolean_operators 

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/grafana/issues/55916

**Special notes for your reviewer**:
1. Run elastic data source by `make devenv sources=elastic`
2. Create simple dashboard with 1 ES panel (any query, default query works)
3. Add ad hoc template variables
4. Apply variables and then jump to explore
5. Check if variables were correctly added

This is how it CORRECTLY WORKS on **this branch**:

https://user-images.githubusercontent.com/30407135/209147111-9d8a0445-6c98-4134-b0fb-ef64e986c368.mov

For comparison, this is how it DOES NOT WORK on **main branch**:

https://user-images.githubusercontent.com/30407135/209147406-62d68879-284d-4027-be94-c62448bb6fc5.mov

### CODE REVIEW

File: public/app/plugins/datasource/elasticsearch/datasource.ts [hunk 1]
```diff
@@ -964,12 +964,43 @@ export class ElasticDatasource
         if (expression.length > 0) {
           expression += ' AND ';
         }
-        expression += `-${action.options.key}:"${action.options.value}"`;
+        expression += `NOT ${action.options.key}:"${action.options.value}"`;
```
Comment (ivanahuckova) [inline public/app/plugins/datasource/elasticsearch/datasource.ts:967]: I changed this to be consistent with other places in elasticsearch code where we are adding `filter out` filter.

Comment (svennergr) [reply public/app/plugins/datasource/elasticsearch/datasource.ts:967]: Haha, I already started with a question why we are changing this. But makes sense, even though https://www.elastic.co/guide/en/elasticsearch/reference/8.5/query-dsl-query-string-query.html#_boolean_operators states that `-` is a `preferred operator`.

Comment (ivanahuckova) [reply public/app/plugins/datasource/elasticsearch/datasource.ts:967]: Nice find. Let me change this back and change my solution to `-` as well.

**Review by svennergr — APPROVED** [review]
Great job - especially on the tests.

Just for the future ours, according to the documentation there might be field names with whitespaces: https://www.elastic.co/guide/en/elasticsearch/reference/8.5/query-dsl-query-string-query.html#_field_names
I think that's an edge case and we can live with that.

**Pull request merged**